dot/parachain/backing/candidate_backing.go

```diff
@@ -5,124 +5,723 @@ package backing
+	PersistedValidationData *parachaintypes.PersistedValidationData
+}
+
+// New creates a new CandidateBacking instance and initialises it with the provided overseer channel.
 func New(overseerChan chan<- any) *CandidateBacking {
 	return &CandidateBacking{
 		SubSystemToOverseer: overseerChan,
 	}
 }
 
 func (cb *CandidateBacking) Run(ctx context.Context, overseerToSubSystem chan any, subSystemToOverseer chan any) error {
-	// TODO: handle_validated_candidate_command
-	// There is one more case where we handle results of candidate validation.
-	// My feeling is that instead of doing it here, we would be able to do that along with processing
-	// other backing related overseer message.
-	// This would become more clear after we complete processMessages function. It would give us clarity
-	// if we need background_validation_rx or background_validation_tx, as done in rust.
-	cb.processMessages()
-	return nil
+	chRelayParentAndCommand := make(chan RelayParentAndCommand)
+	for {
+		select {
+		case rpAndCmd := <-chRelayParentAndCommand:
+			if err := cb.processValidatedCandidateCommand(rpAndCmd); err != nil {
+				logger.Error(err.Error())
+			}
+		case msg := <-cb.OverseerToSubSystem:
+			if err := cb.processOverseerMessage(msg, chRelayParentAndCommand); err != nil {
+				logger.Error(err.Error())
+			}
+		case <-ctx.Done():
+			close(cb.SubSystemToOverseer)
+			close(chRelayParentAndCommand)
+		}
+	}
 }
 
 func (*CandidateBacking) Name() parachaintypes.SubSystemName {
 	return parachaintypes.CandidateBacking
 }
 
-func (cb *CandidateBacking) processMessages() {
-	for msg := range cb.OverseerToSubSystem {
-		// process these received messages by referencing
-		// https://github.com/paritytech/polkadot-sdk/blob/769bdd3ff33a291cbc70a800a3830638467e42a2/polkadot/node/core/backing/src/lib.rs#L741
-		switch msg.(type) {
-		case ActiveLeavesUpdate:
-			cb.handleActiveLeavesUpdate()
-		case GetBackedCandidates:
-			cb.handleGetBackedCandidates()
-		case CanSecond:
-			cb.handleCanSecond()
-		case Second:
-			cb.handleSecond()
-		case Statement:
-			cb.handleStatement()
-		default:
-			logger.Error("unknown message type")
-		}
+// processOverseerMessage processes incoming messages from overseer
+func (cb *CandidateBacking) processOverseerMessage(msg any, chRelayParentAndCommand chan RelayParentAndCommand) error {
+	// process these received messages by referencing
+	// https://github.com/paritytech/polkadot-sdk/blob/769bdd3ff33a291cbc70a800a3830638467e42a2/polkadot/node/core/backing/src/lib.rs#L741
+	switch msg := msg.(type) {
+	case ActiveLeavesUpdate:
+		cb.handleActiveLeavesUpdate()
+	case GetBackedCandidatesMessage:
+		cb.handleGetBackedCandidatesMessage()
+	case CanSecondMessage:
+		cb.handleCanSecondMessage()
+	case SecondMessage:
+		cb.handleSecondMessage()
+	case StatementMessage:
+		return cb.handleStatementMessage(msg.RelayParent, msg.SignedFullStatement, chRelayParentAndCommand)
+	default:
+		return errors.New("unknown message type")
```

Comment:
Passing the message type to the error context could be helpful in debug scenarios
```suggestion
		return fmt.Errorf("unknown message type: %T", msg)
```

Reply:
Can't imagine a situation where we would be interested in value of the message here.

If we reach here, that basically means some problem in code or in parsing the message into right type.

Reply:
> If we reach here, that basically means some problem in code or in parsing the message into right type.

@kishansagathiya  yeah, I agree with that, and that is why I'm suggesting providing more context to the error instead of just returning a plain `unknown message type`.